TWLight/users/models.py

```diff
@@ -357,9 +367,63 @@ def get_wp_groups_display(self):
         else:
             return None
 
+    def authorized_for_bundle(self):
+        user_bundle_authorizations = Authorization.objects.filter(
+            user=self.user, partners__authorization_method=Partner.BUNDLE
+        ).exists()
+        if user_bundle_authorizations:
+            return True
+        else:
+            return False
+
     def get_global_userinfo(self, identity):
         return editor_global_userinfo(identity["username"], identity["sub"], True)
 
+    def update_bundle_authorization(self):
+        """
+        Create or expire this user's bundle authorizations
+        if necessary.
+        The list of partners for the auth will be kept up-to-date
+        elsewhere after initial creation, so no need to worry about
+        updating an existing auth with the latest bundle partner
+        changes here.
+        """
+        # Although we have multiple partners with the BUNDLE authorization
+        # method, we should only ever find one or zero authorizations
+        # for bundle partners.
+        bundle_authorization = Authorization.objects.filter(
```

Comment:
Out of the gate we could grab active authorization(s) so that we do one check for whether it exists or not.